### ISSUE TITLE
[ILM] Remove the "Remove Policy API" in favour of a settings listener

### ISSUE THREAD

**Issue by colings86** [elastic/elasticsearch#34310]
It may be worth removing the "remove policy API following https://github.com/elastic/elasticsearch/pull/34304 and instead registering a Settings listener on the `index.lifecycle.name` setting which will remove the relevant index metadata  from the index when the setting is set to `null`

**Comment by dakrone** [comment]
Tal and I spent some time talking about this. We figured it was a good time to
document some of the issues with doing this without a large change.

The difficulty for this is that when a policy is removed from an index, we want
to remove other various things from the state of the index, so that it returns
to a "clean" state. However, right now there isn't plumbing for invoking a new
cluster state update on a settings change, since the `ClusterStateApplier` is
meant for updating internal data structures prior to a cluster state update
becoming visible, not for making subsequent changes to the cluster state.

There could be a removal followed by a master node failover causing an election
and the new master would silently not remove the settings.
